File: src/main/java/quizapi/model/GameSession.java
package quizapi.model;

import javax.persistence.*;
import java.util.Collection;
import java.util.Set;

@Entity
@Table(name = "game_sessions")
public class GameSession {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", unique = true, nullable = false)
    private Long id;

    @Column(name = "score", nullable = false)
    private Integer score;

    @Column(name = "questions_count_total", nullable = false)
    private Integer questionsCountTotal;

    @Column(name = "questions_count_answered", nullable = false)
    private Integer questionsCountAnswered;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "category_id", nullable = false)
    private Category category;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @ManyToMany(cascade = {CascadeType.ALL}, targetEntity = SessionAnswer.class)
    private Set<SessionAnswer> sessionAnswers;

    public GameSession() {
    }

    public GameSession(Integer score, Integer questionsCountAnswered, Integer questionsCountTotal,  Category category, User user, Set<SessionAnswer> sessionAnswers) {
        this.score = score;
        this.questionsCountTotal = questionsCountTotal;
        this.questionsCountAnswered = questionsCountAnswered;
        this.category = category;
        this.user = user;
        this.sessionAnswers = sessionAnswers;
    }

    public Long getId() {
        return id;
    }

    public Integer getScore() {
        return score;
    }

    public void setScore(Integer score) {
        this.score = score;
    }

    public Category getCategory() {
        return category;
    }

    public void setCategory(Category category) {
        this.category = category;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public Integer getQuestionsCountTotal() {
        return questionsCountTotal;
    }

    public void setQuestionsCountTotal(Integer questionsCountTotal) {
        this.questionsCountTotal = questionsCountTotal;
    }

    public Integer getQuestionsCountAnswered() {
        return questionsCountAnswered;
    }

    public void setQuestionsCountAnswered(Integer questionsCountAnswered) {
        this.questionsCountAnswered = questionsCountAnswered;
    }

    public Set<SessionAnswer> getSessionAnswers() {
        return sessionAnswers;
    }

    public void setSessionAnswers(Set<SessionAnswer> sessionAnswers) {
        this.sessionAnswers = sessionAnswers;
    }

    public void setId(Long id) {
        this.id = id;
    }
}

File: src/main/kotlin/com/shkila/quizapi/model/GameSession.kt
package com.shkila.quizapi.model

import javax.persistence.CascadeType
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.FetchType
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id
import javax.persistence.JoinColumn
import javax.persistence.ManyToMany
import javax.persistence.ManyToOne
import javax.persistence.Table

@Entity
@Table(name = "game_sessions")
data class GameSession(

        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        @Column(name = "id", unique = true, nullable = false)
        var id: Long? = null,

        @Column(name = "score", nullable = false)
        var score: Int = 0,

        @Column(name = "questions_count_total", nullable = false)
        var questionsCountTotal: Int? = null,

        @Column(name = "questions_count_answered", nullable = false)
        var questionsCountAnswered: Int = 0,

        @ManyToOne(fetch = FetchType.EAGER)
        @JoinColumn(name = "category_id", nullable = false)
        var category: Category? = null,

        @ManyToOne(fetch = FetchType.EAGER)
        @JoinColumn(name = "user_id", nullable = false)
        var user: User? = null,

        @ManyToMany(cascade = [CascadeType.ALL], targetEntity = SessionAnswer::class)
        var sessionAnswers: MutableCollection<SessionAnswer>? = null
)
